### ISSUE TITLE
Get Hi Res Pics

### ISSUE THREAD

**Issue by ghost** [timdewhirst/blip#1]
PASS  return_image_urls 

GET image_urls -------   hires

**Comment by timdewhirst** [comment]
Hey,

Yeah, I saw that but unfortunately it's blank...

Tim

**Comment by lumpysimon** [comment]
Did you have any luck figuring this one out? 

I'm writing an importer to get all my entries into WordPress. I've tried setting the `return_image_urls` parameter to `1`, but this still only returns the low res and std res images - the high res and original fields are always blank. I get the same result when I use the console to test it.

![console-test](https://cloud.githubusercontent.com/assets/625372/6658148/ad6452e6-cb58-11e4-83df-427c3b60d42d.png)

**Comment by ghost** [comment]
does only some apps have access? .... has anyone tried hacking android/apple app for tokens?

**Comment by StephanieKemna** [comment]
@NanneHuiges may have some ideas?

**Comment by NanneHuiges** [comment]
you mean access tokens? You can just get one from blipfoto via https://www.polaroidblipfoto.com/developer/api so no hacking is needed :)

For the emtpy result: maybe different auth is needed for the image_urls, see the reference for user and app authentication. 

btw: be advised that this is not a supported use-case (it is explicitly forbidden to use the API for bulk collection I believe, see rule 7 on https://www.polaroidblipfoto.com/developer/rules ) so I would run it with care, if at all. I'd suggest you at least limit your requests :)

**Comment by lumpysimon** [comment]
I'm using the access token method in my WordPress plugin, it still doesn't return hi res images.

I saw that clause in the API, but I'd say that exporting your **own** images doesn't fall under that (I interpret it as referring to bulk collection of images from across multiple users). Given that the future of Blipfoto is unclear, it sounds very reasonable to me to allow users to export their own data.

**Comment by NanneHuiges** [comment]
I agree with you that it would be reasonable, but I disagree with you about the clause: it has no such exception. I'm not saying that's right or not, I'm just saying I think blipfoto might interpret it as such, so your API key might get blocked. Anyway, I just wanted to mention it so you can take it into account :)

What kind of access-token are you using? I believe that originals aren't visible to viewers (e.g. I can not see the original of your shots) so I suppose you'd need User-auth (with your own username), and not App-auth.

That would not be enough to explain why there is no hi-res though ;(

**Comment by lumpysimon** [comment]
Yes, you're right, 'reasonable' or 'right' are interpretations, not something we can assume to be allowed.

However, because I've done it as a web app, whereby each user creates their own app on Blipfoto and then enters their ID, secret & access token into their website, it is a unique app just for their user account (it works in exactly the same way that Twitter apps work if you want to show your tweets on your own website). This means there isn't a centralised app key that can be blocked. Blipfoto could reduce the rate limits, or cut off the API altogether, but they couldn't block at the app level.

This means authorisation is done at the user level, so if that's not showing original images then I imagine it means there's no way at all to retrieve them.

**Comment by NanneHuiges** [comment]
Hmpf. then I'm all out of ideas for the api.

Rude method would be to retrieve the entry-url and scrape it for the download button that's on the page. That might work?

**Comment by lumpysimon** [comment]
Ah, clever thinking! I might give that a go.
